Jenkinsfile

```diff
@@ -1,209 +1,211 @@
 
-                                                        ctest -R "." -E "(_app_|_unit_|_module_)" -j4 -VV
-                                                        source deactivate
+def setGitHubBuildStatus(status) {
+    def repoUrl = getRepoUrl()
+    def commitSha = getCommitSha()
+
+    step([
+        $class: 'GitHubCommitStatusSetter',
+        reposSource: [$class: "ManuallyEnteredRepositorySource", url: repoUrl],
+        commitShaSource: [$class: "ManuallyEnteredShaSource", sha: commitSha],
+        errorHandlers: [[$class: 'ShallowAnyErrorHandler']],
+        statusResultSource: [
+          $class: 'ConditionalStatusResultSource',
+          results: [
+            [$class: 'BetterThanOrEqualBuildResult', result: 'SUCCESS', state: 'SUCCESS', message: status],
+            [$class: 'BetterThanOrEqualBuildResult', result: 'FAILURE', state: 'FAILURE', message: status],
+            [$class: 'AnyBuildResult', state: 'FAILURE', message: 'Loophole']
+          ]
+        ]
+    ])
+}
 
-                                                    """
-                                                }
-                                                catch(e) {
-                                                    build_ok = false
-                                                    errors.add(env.STAGE_STATUS)
-                                                    println e.toString()
-                                                }
-                                            }
-                                        }
+pipeline {
+  agent any
+  stages {
+    stage("CI") {
+      steps { 
+        script {
+          def groovy_utilities = load "${pwd()}/groovy_utilities.groovy" 
+      
+          def cmakeFlags = [
+              "-DJP2KFLAG=ON",
+              "-DKAKADU_INCLUDE_DIR=/isisData/kakadu",
+              "-Dpybindings=OFF",
+              "-DCMAKE_BUILD_TYPE=RELEASE"
+          ]
+
+          def build_ok = true
+          def errors = []
+          def labels = ['CentOS', 'Fedora', 'Ubuntu', 'Mac'] // labels for Jenkins node types we will build on
+          def builders = [:] 
+          
+          println("Using Labels: " + labels)
+
+          for (x in labels) {
+            def label = x
+            def lower_label = label.toLowerCase()
+        
+            builders[lower_label] = {
+              node(lower_label) {
+              
+              def condaPath = ""
+              def isisEnv = [
+                "ISIS3DATA=/isisData/data",
+                "ISIS3TESTDATA=/isisData/testData",
+                "ISIS3MGRSCRIPTS=/isisData/data/isis3mgr_scripts",
+              ]
+              stage (label) {
+                  sh 'git config --global http.sslVerify false'
+                  checkout scm
+                  isisEnv.add("ISISROOT=${pwd()}/build")
+                  cmakeFlags.add("-DCMAKE_INSTALL_PREFIX=${pwd()}/install")
```

Comment:
You can use a relative path for the install prefix so that it can be defined with the other CMAKE flags.

Reply:
It originally used relative paths which is a bit messier (e.g. ../../install). This is a bit more explicit that this is different on a per OS basis. If anything I'd rather make it more explicit with $WORKSPACE/install. But I'll leave that as a separate issue.